test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIntegrationTest.cs

```diff
@@ -552,6 +552,31 @@ public async Task Build_WithP2P_WorksWhenBuildProjectReferencesIsDisabled()
             Assert.BuildPassed(result);
         }
 
+        [Fact]
+        [InitializeTestProject("AppWithP2PReference", additionalProjects: new[] { "ClassLibrary", "ClassLibraryMvc21" })]
+        public async Task Build_WithP2P_Referencing21Project_Works()
+        {
+            // Verifies building with different versions of Razor.Tasks works. Loosely modeled after the repro
+            // scenario listed in https://github.com/Microsoft/msbuild/issues/3572
+            var additionalProjectContent = @"
+<ItemGroup>
+  <ProjectReference Include=""..\ClassLibraryMvc21\ClassLibraryMvc21.csproj"" />
+</ItemGroup>
+";
+            AddProjectFileContent(additionalProjectContent);
+
+            var result = await DotnetMSBuild(target: default);
+
+            Assert.BuildPassed(result);
```

Comment:
This fails without the fix like so:

```
  ClassLibrary -> C:\Users\Pranav\AppData\Local\Temp\Razor\q4mzeswy.5fw\ClassLibrary\bin\Debug\netstandard2.0\ClassLibrary.dll
  ClassLibrary -> C:\Users\Pranav\AppData\Local\Temp\Razor\q4mzeswy.5fw\ClassLibrary\bin\Debug\netstandard2.0\ClassLibrary.Views.dll
  ClassLibraryMvc21 -> C:\Users\Pranav\AppData\Local\Temp\Razor\q4mzeswy.5fw\ClassLibraryMvc21\bin\Debug\netstandard2.0\ClassLibraryMvc21.dll
  ClassLibraryMvc21 -> C:\Users\Pranav\AppData\Local\Temp\Razor\q4mzeswy.5fw\ClassLibraryMvc21\bin\Debug\netstandard2.0\ClassLibraryMvc21.Views.dll
D:\work\Razor\src\Microsoft.AspNetCore.Razor.Design\build\netstandard2.0\Microsoft.AspNetCore.Razor.Design.CodeGeneration.targets(69,5): error MSB4062: The "Microsoft.AspNetCore.Razor.Tasks.RazorTagHelper" task could not be loaded from the assembly D:\work\Razor\src\Microsoft.AspNetCore.Razor.Design\bin\Debug\\tasks\netstandard2.0\Microsoft.AspNetCore.Razor.Tasks.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Users\Pranav\AppData\Local\Temp\Razor\q4mzeswy.5fw\AppWithP2PReference\AppWithP2PReference.csproj]



Stack Trace:
   at Microsoft.AspNetCore.Razor.Design.IntegrationTests.Assert.BuildPassed(MSBuildResult result) in D:\work\Razor\test\Microsoft.AspNetCore.Razor.Design.Test\IntegrationTests\Assert.cs:line 29
```

Reply:
Btw, notice how ClassLibrary and ClassLibraryMvc21 built just fine even though they used conflicting versions. The assembly loading seems to be a problem only when projects reference each other